File: Programacion Modulo Compras/SAE-COMPRAS/CapaPresentacion/FrmOrdenEntregadoDetalles.vb
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms
Imports CapaNegocio

Public Partial Class FrmOrdenEntregadoDetalles
	Inherits Form
	Private _owner As Ordenes

	Public Sub New()
		InitializeComponent()
	End Sub

	Public Sub New(owner As Ordenes)
		_owner = owner

		InitializeComponent()

		Me.labelNumeroPedido.Text = _owner.ObtenerSeleccionPedidos().Cells("CÓDIGO").Value.ToString()
		Me.textBoxCliente.Text = _owner.ObtenerSeleccionPedidos().Cells("PROVEEDOR").Value.ToString()
		Me.textBoxTransporte.Text = _owner.ObtenerSeleccionPedidos().Cells("TRANSPORTE").Value.ToString()
		Me.textBoxEmpleado.Text = _owner.ObtenerSeleccionPedidos().Cells("RESPONSABLE").Value.ToString()
		Me.textBoxFechaOrdenado.Text = Convert.ToDateTime(_owner.ObtenerSeleccionPedidos().Cells("ORDENADO").Value).ToString("dd/MM/yyyy")
		Me.textBoxFechaEntregado.Text = Convert.ToDateTime(_owner.ObtenerSeleccionPedidos().Cells("ENTREGADO").Value).ToString("dd/MM/yyyy")
		

		Me.textBoxCostoEnvio.Text = Convert.ToDecimal(_owner.ObtenerSeleccionPedidos().Cells("COSTO DE ENVÍO").Value).ToString("0.00#")
		Me.dataGridViewProductos.DataSource = NOrdenDetalles.MostrarOrdenDetalles(Convert.ToInt32(_owner.ObtenerSeleccionPedidos().Cells(1).Value))
		Me.dataGridViewProductos.Columns(4).DefaultCellStyle.Format = "0.00# '$'"
		Me.dataGridViewProductos.Columns(4).DefaultCellStyle.Alignment = DataGridViewContentAlignment.MiddleRight
		Me.dataGridViewProductos.Columns(5).DefaultCellStyle.Format = "0.00# '%'"
		Me.dataGridViewProductos.Columns(5).DefaultCellStyle.Alignment = DataGridViewContentAlignment.MiddleRight
	End Sub

	Private Sub buttonSalir_Click(sender As Object, e As EventArgs)
		Me.Close()
	End Sub

	Private Sub dataGridViewProductos_RowPrePaint(sender As Object, e As DataGridViewRowPrePaintEventArgs)
		Me.dataGridViewProductos.Rows(e.RowIndex).Cells(0).Value = (e.RowIndex + 1).ToString()
	End Sub

End Class

File: Programacion Modulo Compras/SAE-COMPRAS/CapaPresentacion/FrmOrdenEntregadoDetalles.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using CapaNegocio;

namespace CapaPresentacion
{
    public partial class FrmOrdenEntregadoDetalles : Form
    {
        Ordenes _owner;

        public FrmOrdenEntregadoDetalles()
        {
            InitializeComponent();
        }
        public FrmOrdenEntregadoDetalles(Ordenes owner)
        {
            _owner = owner;

            InitializeComponent();

            this.labelNumeroPedido.Text = _owner.ObtenerSeleccionPedidos().Cells["CÓDIGO"].Value.ToString();
            this.textBoxCliente.Text = _owner.ObtenerSeleccionPedidos().Cells["PROVEEDOR"].Value.ToString();
            this.textBoxFechaOrdenado.Text = Convert.ToDateTime(_owner.ObtenerSeleccionPedidos().Cells["ORDENADO"].Value).ToString("dd/MM/yyyy");
            this.textBoxFechaEntregado.Text = Convert.ToDateTime(_owner.ObtenerSeleccionPedidos().Cells["ENTREGADO"].Value).ToString("dd/MM/yyyy"); ;
            this.dataGridViewProductos.DataSource = NOrdenDetalles.MostrarOrdenDetalles(Convert.ToInt32(_owner.ObtenerSeleccionPedidos().Cells[1].Value));
            this.dataGridViewProductos.Columns[4].DefaultCellStyle.Format = "0.00# '$'";
            this.dataGridViewProductos.Columns[4].DefaultCellStyle.Alignment = DataGridViewContentAlignment.MiddleRight;
            this.dataGridViewProductos.Columns[5].DefaultCellStyle.Format = "0.00# '%'";
            this.dataGridViewProductos.Columns[5].DefaultCellStyle.Alignment = DataGridViewContentAlignment.MiddleRight;
        }

        private void pictureBox6_Click(object sender, EventArgs e)
        {
            this.Close();
        }

        private void dataGridViewProductos_RowPrePaint(object sender, DataGridViewRowPrePaintEventArgs e)
        {
            this.dataGridViewProductos.Rows[e.RowIndex].Cells[0].Value = (e.RowIndex + 1).ToString();
        }
    }
}
